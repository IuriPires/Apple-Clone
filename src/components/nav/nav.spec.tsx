import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

import Nav from ".";

const NAV_LINKS = [
  { href: "/store", text: "Store" },
  { href: "/mac", text: "Mac" },
  { href: "/ipad", text: "iPad" },
  { href: "/iphone", text: "iPhone" },
  { href: "/watch", text: "Watch" },
  { href: "/vision", text: "Vision" },
  { href: "/airpods", text: "AirPods" },
  { href: "/tv-and-home", text: "TV & Home" },
  { href: "/entertainment", text: "Entertainment" },
  { href: "/accessories", text: "Accessories" },
  { href: "/support", text: "Support" },
];

describe("", () => {
  test("Should render Nav correctly", () => {
    const { baseElement } = render(<Nav linksList={NAV_LINKS} />);

    expect(baseElement).toMatchSnapshot();
  });

  it.each(NAV_LINKS)(
    "Should display the correct text $text for each link",
    ({ text }) => {
      const { baseElement } = render(<Nav linksList={NAV_LINKS} />);
      expect(baseElement).toHaveTextContent(text);
    }
  );
});
