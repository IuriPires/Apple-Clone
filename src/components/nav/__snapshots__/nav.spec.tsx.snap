// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Should render Nav correctly 1`] = `
<body>
  <div>
    <nav
      class="w-full flex gap-5 items-center bg-[#161617CC] px-3 py-3 justify-between lg:justify-center"
    >
      <span
        class="pr-3 cursor-pointer"
      >
        <svg
          class="text-lg"
          color="white"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: white;"
          viewBox="0 0 384 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M318.7 268.7c-.2-36.7 16.4-64.4 50-84.8-18.8-26.9-47.2-41.7-84.7-44.6-35.5-2.8-74.3 20.7-88.5 20.7-15 0-49.4-19.7-76.4-19.7C63.3 141.2 4 184.8 4 273.5q0 39.3 14.4 81.2c12.8 36.7 59 126.7 107.2 125.2 25.2-.6 43-17.9 75.8-17.9 31.8 0 48.3 17.9 76.4 17.9 48.6-.7 90.4-82.5 102.6-119.3-65.2-30.7-61.7-90-61.7-91.9zm-56.6-164.2c27.3-32.4 24.8-61.9 24-72.5-24.1 1.4-52 16.4-67.9 34.9-17.5 19.8-27.8 44.3-25.6 71.9 26.1 2 49.9-11.4 69.5-34.3z"
          />
        </svg>
      </span>
      <div
        class="gap-5 items-center hidden lg:flex  lg:visible lg:justify-center"
      >
        <a
          class="font-light text-sm text-slate-200 hover:text-white transition-all"
          href="/store"
        >
          Store
        </a>
        <a
          class="font-light text-sm text-slate-200 hover:text-white transition-all"
          href="/mac"
        >
          Mac
        </a>
        <a
          class="font-light text-sm text-slate-200 hover:text-white transition-all"
          href="/ipad"
        >
          iPad
        </a>
        <a
          class="font-light text-sm text-slate-200 hover:text-white transition-all"
          href="/iphone"
        >
          iPhone
        </a>
        <a
          class="font-light text-sm text-slate-200 hover:text-white transition-all"
          href="/watch"
        >
          Watch
        </a>
        <a
          class="font-light text-sm text-slate-200 hover:text-white transition-all"
          href="/vision"
        >
          Vision
        </a>
        <a
          class="font-light text-sm text-slate-200 hover:text-white transition-all"
          href="/airpods"
        >
          AirPods
        </a>
        <a
          class="font-light text-sm text-slate-200 hover:text-white transition-all"
          href="/tv-and-home"
        >
          TV & Home
        </a>
        <a
          class="font-light text-sm text-slate-200 hover:text-white transition-all"
          href="/entertainment"
        >
          Entertainment
        </a>
        <a
          class="font-light text-sm text-slate-200 hover:text-white transition-all"
          href="/accessories"
        >
          Accessories
        </a>
        <a
          class="font-light text-sm text-slate-200 hover:text-white transition-all"
          href="/support"
        >
          Support
        </a>
      </div>
      <div
        class="flex gap-3 items-center"
      >
        <span
          class="pl-3 cursor-pointer"
        >
          <svg
            class="text-lg text-slate-200 hover:text-white"
            fill="currentColor"
            height="1em"
            id="search"
            stroke="currentColor"
            stroke-width="1.1"
            version="1.1"
            viewBox="0 0 24 24"
            width="1em"
            x="0px"
            xmlns="http://www.w3.org/2000/svg"
            y="0px"
          >
            <g>
              <path
                d="M20.031,20.79c0.46,0.46,1.17-0.25,0.71-0.7l-3.75-3.76c1.27-1.41,2.04-3.27,2.04-5.31
		c0-4.39-3.57-7.96-7.96-7.96s-7.96,3.57-7.96,7.96c0,4.39,3.57,7.96,7.96,7.96c1.98,0,3.81-0.73,5.21-1.94L20.031,20.79z
		 M4.11,11.02c0-3.84,3.13-6.96,6.96-6.96c3.84,0,6.96,3.12,6.96,6.96c0,3.84-3.12,6.96-6.96,6.96C7.24,17.98,4.11,14.86,4.11,11.02
		z"
              />
            </g>
          </svg>
        </span>
        <span
          class="pl-3 cursor-pointer"
        >
          <svg
            class="text-lg text-slate-200 hover:text-white"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="1.1"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M80 176a16 16 0 0 0-16 16v216c0 30.24 25.76 56 56 56h272c30.24 0 56-24.51 56-54.75V192a16 16 0 0 0-16-16zm80 0v-32a96 96 0 0 1 96-96h0a96 96 0 0 1 96 96v32"
              fill="none"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="32"
            />
          </svg>
        </span>
        <span
          class="pl-3 lg:hidden cursor-pointer"
        >
          <svg
            class="text-lg text-slate-200 hover:text-white"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 15 15"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M1.5 3C1.22386 3 1 3.22386 1 3.5C1 3.77614 1.22386 4 1.5 4H13.5C13.7761 4 14 3.77614 14 3.5C14 3.22386 13.7761 3 13.5 3H1.5ZM1 7.5C1 7.22386 1.22386 7 1.5 7H13.5C13.7761 7 14 7.22386 14 7.5C14 7.77614 13.7761 8 13.5 8H1.5C1.22386 8 1 7.77614 1 7.5ZM1 11.5C1 11.2239 1.22386 11 1.5 11H13.5C13.7761 11 14 11.2239 14 11.5C14 11.7761 13.7761 12 13.5 12H1.5C1.22386 12 1 11.7761 1 11.5Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </div>
    </nav>
  </div>
</body>
`;
